pkg/configs/client/client.go

```diff
@@ -163,3 +119,52 @@ func (c ConfigsResponse) GetLatestConfigID() configs.ID {
 	}
 	return latest
 }
+
+// ListAllRuleGroups polls the configdb server and returns the updated rule groups
+func (c *ConfigDBClient) ListAllRuleGroups(ctx context.Context) (map[string]rules.RuleGroupList, error) {
```

Comment:
Did we drop the since functionality for a reason?  Any concern for extremely large rule sets that this will return very large amounts of data.  Otherwise I like removing it for simplicity.

Reply:
Should this method continue returning `map[string]configs.VersionedRulesConfig` and the translation to the new type occur in `pkg/ruler/rules`?

Reply:
I would say `since` was for large numbers of tenants, regardless of size of rule set.  And I am a bit nervous about dropping it.

Reply:
I do understand the concern. The main reason since was dropped is because the entire ruleset must be hashed when generating mapped files. Otherwise changes to the ring will not be reflected in the scheduled evaluations. However, polling such a large payload is not ideal. One option would be for the config client to cache the previous response and have an internal usage of the since variable. That way it can keep an up-to-date set of the active rules configs and only poll for changes.

Reply:
Per discussions I'm re-adding `since` support now.